### PR TITLE
Move Grafana CRUD commands to AAZ

### PR DESCRIPTION
**Pull request by ABZhang0**
### Related command
`az grafana create`, `az grafana update`, `az grafana list`, `az grafana show`, `az grafana delete`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by kairu-ms** [comment]
Please check whether those breaking change are expexted 
image

File: src/amg/azext_amg/custom.py [hunk 1]
```diff
@@ -25,55 +28,108 @@
...
+                 'Once provided, CLI won\'t make the current logged-in user as Grafana Admin',
+        )
+        args_schema['principal_ids'].Element = AAZStrArg()
+        args_schema['identity']._registered = False  # pylint: disable=protected-access
+
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if args.skip_role_assignments and args.principal_ids:
```
Comment (kairu-ms) [inline src/amg/azext_amg/custom.py:57]:
```suggestion
        if has_value(args.skip_role_assignments) and has_value(args.principal_ids):
```

File: src/amg/azext_amg/custom.py [hunk 2]
```diff
@@ -25,55 +28,108 @@
...
+            raise ArgumentUsageError("--skip-role-assignments | --assignee-object-ids")
+
+        if not args.skip_system_assigned_identity:
```
Comment (kairu-ms) [inline src/amg/azext_amg/custom.py:60]:
```suggestion
        if not has_value(args.skip_system_assigned_identity):
```

File: src/amg/azext_amg/custom.py [hunk 3]
```diff
@@ -25,55 +28,108 @@
...
+            args.identity = {"type": "SystemAssigned"}
+
+        process_grafana_create_namespace(self.ctx, self.ctx.args)
+
+    # override the output method to create role assignments after instance creation
+    def _output(self, *args, **kwargs):
```
Comment (kairu-ms) [inline src/amg/azext_amg/custom.py:66]: It should be in `post_operations` function instead of `_output`

Comment (ABZhang0) [reply src/amg/azext_amg/custom.py:66]: Unfortunately `post_operations` does not have ``self.ctx.vars.instance`` available to reference, and `post_instance_create`/`post_instance_update`, which does have that, is only available to override in subcommands. I found this way to achieve what was needed while still using the recommended inheritance method of customization. Is this acceptable?

Comment (kairu-ms) [reply src/amg/azext_amg/custom.py:68]: OK, that's fine. It's because it's a LRO and the instance will created only when LRO pulling finished.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**